polygeist/lib/polygeist/Passes/AffineReduction.cpp

```diff
@@ -159,17 +155,12 @@ class AffineForReductionIter : public OpRewritePattern<AffineForOp> {
       llvm::dbgs() << "\n";
     });
 
-    WalkResult Result = ForOp.getBody()->walk([&](Operation *Op) {
-      // We are only interested in affine load operations.
-      if (!isa<AffineLoadOp>(Op))
-        return WalkResult::advance();
-
-      auto Load = cast<AffineLoadOp>(Op);
+    WalkResult Result = ForOp.getBody()->walk([&](AffineLoadOp Load) {
       LLVM_DEBUG(llvm::dbgs() << "Load: " << Load << "\n");
 
       // Skip the load if any of its subscript indices are the loop induction
       // variable (i.e. the load is not loop invariant).
-      SmallVector<Value, 4> Indices(Load.getIndices());
+      const SmallVector<Value, 4> Indices = Load.getIndices();
```

Comment:
`const Operation::operand_range Indices = Load.getIndices();`?

Reply:
:+1: , that or `const auto` would prevent constructing a new `SmallVector` object and possible allocation.

Reply:
done